### PR TITLE
Replace Persona with Auth0.

### PR DESCRIPTION
**Pull request by akatsoulas**
Tested the following:
* An existing user can login.
* A new (invited) user can create a new profile.
* The new user is blocked until the profile is created
* A vouched Mozillian can login
* A user that does not match the above scenarios gets an error.

### CODE REVIEW

**Comment by johngian** [comment]
Please remove ``django-browserid`` from requirements.

**Comment by johngian** [comment]
Please remove references to browserid from the documentation

**Comment by akatsoulas** [comment]
@johngian pr updated

**Comment by johngian** [comment]
```
$ grep -R "BROWSERID_USERNAME_ALGO"
remo/profiles/views.py:USERNAME_ALGO = getattr(settings, 'BROWSERID_USERNAME_ALGO',
remo/profiles/forms.py:USERNAME_ALGO = getattr(settings, 'BROWSERID_USERNAME_ALGO',
remo/base/backend.py:USERNAME_ALGO = getattr(settings, 'BROWSERID_USERNAME_ALGO',
```

File: remo/settings/base.py [hunk 1]
```diff
@@ -204,15 +198,14 @@ def COMPRESS_JINJA2_GET_ENVIRONMENT():
 PASSWORD_HASHERS = get_password_hashers(BASE_PASSWORD_HASHERS, HMAC_KEYS)
 
 AUTHENTICATION_BACKENDS = [
-    'django_browserid.auth.BrowserIDBackend',
-    'django.contrib.auth.backends.ModelBackend'
+    'django.contrib.auth.backends.ModelBackend',
```
Comment (johngian) [inline remo/settings/base.py]: Is there a reason why you are changing the order here?

File: remo/settings/local.py-docker-dist [hunk 2]
```diff
@@ -8,7 +8,8 @@ DOMAIN = "127.0.0.1"
 PROTOCOL = "http://"
 PORT = 8000
 
-SITE_URL = '%s%s:%d' % (PROTOCOL, DOMAIN, PORT)
+SITE_URL = 's%s:%d' % (PROTOCOL, DOMAIN, PORT)
```
Comment (johngian) [inline remo/settings/local.py-docker-dist]: It looks like you are missing an ``%s`` in the start of the string.

File: requirements/prod.txt [hunk 3]
```diff
@@ -218,3 +215,8 @@ hashin==0.6.0 \
     --hash=sha256:7563490ec8c9c361e48c7623898b3585143d379714a03bee4c3d52fb13e85086
 django-import-export==0.4.5 \
     --hash=sha256:0406f781a771c39e325474df09644a5cea5188dfac7ecace4f32d664920abd00
+https://github.com/mozilla/mozilla-django-oidc/archive/0.1.2.zip#egg=mozilla_django_oidc \
```
Comment (johngian) [inline requirements/prod.txt]: Please fix ``egg`` definition to allow version upgrades.
It should be:

``egg=mozilla-django-oidc==0.1.2``

Comment (johngian) [inline requirements/prod.txt]: Also please bump to latest version.

Comment (johngian) [inline requirements/prod.txt]: This means you also need to upgrade the rest of the ``mozilla-django-oidc`` dependencies.

**Comment by akatsoulas** [comment]
@johngian pr updated

File: remo/settings/base.py [hunk 4]
```diff
@@ -3,9 +3,9 @@
 import logging
 import os
 
-from django.utils.functional import lazy
 from django_jinja.builtins import DEFAULT_EXTENSIONS
 from django_sha2 import get_password_hashers
+from django.utils.functional import lazy
```
Comment (johngian) [inline remo/settings/base.py]: Django dependencies before third party.

**Comment by johngian** [comment]
Code looks OK. Let's merge it to see what happens on dev.

**Pull request merged**